pkg/collectors/dev_info_collector.go

```diff
@@ -52,9 +52,11 @@ func (ptpDev *DevInfoCollector) monitorErroredPolls() {
 		case <-ptpDev.quit:
 			return
 		case <-ptpDev.erroredPolls:
-			ptpDev.requiresFetch <- true
+			if len(ptpDev.requiresFetch) == 0 {
+				ptpDev.requiresFetch <- true
+			}
 		default:
-			time.Sleep(time.Millisecond)
+			time.Sleep(time.Microsecond)
```

Comment:
Consuming the entire channel would take time and we might continue generating errors during that time so that would block following refreshes which we might want. 

So I just thought refresh more to churn thought them and ignore until we've fetched again (which would make the `requiresFetch` empty). It might mean we get more attempted refreshes overall but I'd prefer to air on the side of more refreshes than less.

Reply:
It will only fetch when it goes to announce again which would rate limit the fetches to the announcement rate. During that time this func will be consume all the events it can. 

Perhaps we could look at a separate refresh rate which scales with consecutive refreshes or something that.

Reply:
In the event this channel becomes full, it would have a cascading block back to the collector once poll results becomes full.